### PR TITLE
Implementation to get status of Workflow from WorkflowContext.

### PR DESCRIPTION
**Pull request by sagarkapare**
JIRA - https://issues.cask.co/browse/CDAP-4075

### CODE REVIEW

File: cdap-unit-test/src/test/java/co/cask/cdap/test/app/WorkflowStatusTestApp.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
+package co.cask.cdap.test.app;
```
Comment (anwar6953) [inline cdap-unit-test/src/test/java/co/cask/cdap/test/app/WorkflowStatusTestApp.java]: dont need license headers?

Comment (sagarkapare) [inline cdap-unit-test/src/test/java/co/cask/cdap/test/app/WorkflowStatusTestApp.java]: missed that. added

**Comment by sagarkapare** [comment]
Build is green.

**Comment by anwar6953** [comment]
Code changes LGTM.
Has the API been reviewed/confirmed?

**Comment by sagarkapare** [comment]
Spoke with @chtyim and marking the new api as `Beta`. Since build already passed squashing and merging.

**Pull request merged**